Update documentation for 'file' module to include 'diff_peek'.

gerph opened the pull request:
diff_peek and validate are options that are for internal use - they shouldn't be used by end users.
So we document them in the code as such.

gerph committed:
Update documentation for 'file' module to include 'diff_peek'.

The 'diff_peek' option isn't documented at all, and provides a
rudimentary check that the content isn't binary. Documentation is
added to explain the option.

The 'validate' option has a declaration, but isn't implemented.
Therefore it may as well be removed from the module.

bcoca commented:
diff_peek is not there for users, but for internal checks in the action plugins. 

The validate option is also there for compatibility with template and copy modules, as they use the file module behind the scenes.

gerph commented:
Ah, right; that wasn't clear.
I can think of 3 ways to capture that information...
1. Add some comments near the AnsibleModule() creation to indicate their internal use.
2. Add a new property to the dictionary for the options in the form 'internal="For use by template and copy which delegate their work to this module"', which wouldn't be used, but could identify the options in a way that /could/ be used in the future.
3. Include in the documentation section for those options 'Internal Use Only'.

(there is also "option 4: do nothing" but I believe that if there are reasons for something that aren't obvious, you should explain yourself)

What do you think? I personally like option 2, as that's the most programatic way of explaining things, and doesn't interfere with anything else.

bcoca commented:
currently i'm leaning for 1, but 2 might make a lot of sense when we move to using module specs to 'autodocument' more.

gerph commented:
I'll go with 1, and let other people argue on how it might be more programatically used.

gerph committed:
Add documentation to 'file' AnsibleModule definition for internals.

The parameters 'diff_peek' and 'validate' are not expected to be used
by users. They are internal. To make it clear, this change adds the
comments 'Internal use only' to each of those definitions to make
it clear that they are actually used, just not by end-users.

gerph commented:
How's that look?

bcoca commented:
lgtm
